### ISSUE TITLE
Incorrect macro-error on Rust By Example code

### ISSUE THREAD

**Issue by matthew-ch** [rust-analyzer/rust-analyzer#6900]
I'm using rust-analyzer extension on VS code. It shows error on the Rust By Example code https://doc.rust-lang.org/stable/rust-by-example/macros/dry.html

Specifically, it says `unexpected token in input rust-analyzer(macro-error)` for the following code:

```rust
op!(add_assign, Add, +=, add);
op!(mul_assign, Mul, *=, mul);
op!(sub_assign, Sub, -=, sub);
```

If I add quotes to the operators, i.e. `+=` becomes `"+="`, the error would be gone.

But I can compile and run the original code without problem. So I'm guessing analyzer gives incorrect error.

rust-analyzer version: v0.2.416

**Comment by edwin0cheng** [comment]
Reproduction:

```rust
macro_rules! op {
    ($op:tt, $a:ident) => {
        fn f() {}
    };
}

op!(+=, a);

fn main() {
    f();
}
```

**Comment by lnicola** [comment]
Duplicate of #6692?

**Comment by edwin0cheng** [comment]
I think so, let's close this one in favor of the that.
